src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -59,6 +59,23 @@ void MemoryCounter::update_peak(size_t size, size_t cnt) {
   }
 }
 
+void MallocMemorySnapshot::copy_to(MallocMemorySnapshot* s) {
+  // Need to make sure that mtChunks don't get deallocated while the
+  // copy is going on, because their size is adjusted using this
+  // buffer in make_adjustment().
```

Comment:
Since you are moving this, could you change the comment a bit to make clear it only refers to the ThreadCritical, not the rest of the function? E.g. "Need to lock to make sure...."

Reply:
Done.